src/pyseekdb/client/kernel_errors.py

```
@@ -0,0 +1,183 @@
+            return True
+    return False
+
+
+def _is_namespace_dropping_error(exc: BaseException) -> bool:
+    """Whether the kernel reports a namespace is still being dropped."""
+    text = _exception_text(exc).lower()
+    return (
+        _has_ob_error_code(exc, 4109)
+        or "namespace is dropping" in text
+        or "namespace is in drop" in text
+        or "being dropped" in text and "namespace" in text
```

Ruff (0.15.18)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)